File: ProjectTest/share/Network/Protocol/ProtocolEvents.cs
using System;

namespace Share.Network.Protocol
{
    public abstract class ProtocolEvents<T>
    {

        public readonly string eventName;

        public ProtocolEvents(string name)
        {
            eventName = name;
        }

        public ProtocolNetwork GetProtocol()
        {
            ProtocolNetwork pn = new Protocol<T>
            {
                EventName = eventName
            };
            return pn;
        }

    }
}

File: ClientJava/network/src/main/java/network/protocol/ProtocolEvents.java
package network.protocol;

import network.share.DataListener;
import network.share.DataListenerTCP;

public class ProtocolEvents<T> {

    public String eventName;

    public ProtocolEvents(String name){
        eventName = name;
    }

    public Protocol getProtocol(Class<T> co, DataListener<T> listener){
        return new Protocol<T>(co, eventName, listener);
    }

    public Protocol getProtocol(Class<T> co, DataListenerTCP<T> listener){
        return new Protocol<T>(co, eventName, listener);
    }

}
